lib/elixir/pages/Deprecations.md

```diff
@@ -0,0 +1,57 @@
+# Deprecations
+
+## Deprecation policy
+
+Elixir deprecations happens in 3 steps:
+
+  1. The feature is soft-deprecated. It means both CHANGELOG and documentation must list the feature as deprecated but no warning is effectively emitted by running the code. There is no requirement to soft-deprecate a feature.
+
+  2. The feature is effectively deprecated by emitting warnings on usage. In order to deprecate a feature, the proposed alternative MUST exist for AT LEAST two versions. For example, `Enum.uniq/2` was soft-deprecated in favor of `Enum.uniq_by/2` in Elixir v1.1. This means a deprecation warning may only be emitted by Elixir v1.3 or later.
+
+  3. The feature is removed. This can only happen on major releases. This means deprecated features in Elixir v1.x shall only be removed by Elixir v2.x.
+
+
+## Table of deprecated features: modules, functions, behaviours, protocols, typespecs, attributes and arguments
+
+Deprecated feature        | Deprecated in | Replaced by             | Replacement available since
+:------------------------ | :------------ | :---------------------- | :---------------------------
+`GenEvent` module         | [v1.5]    | `Supervisor` and `GenServer`; `GenStage`; `:gen_event` | v1.0 (`Supervisor and GenServer`); v1.3 (GenStage); OTP v17.0 (`:gen_server`)
+`()` to mean `nil`        | [v1.5]    | `nil`                       | v1.0
+`@compile {:parse_transform, _}` in `Module` | [v1.5] | __???__     | __???__
+`Atom.to_char_list/1`     | [v1.5]    | `Atom.to_charlist/1`        | v1.3
+`Float.to_char_list/1`    | [v1.5]    | `Float.to_charlist/1`       | v1.3
+`Integer.to_char_list/1`  | [v1.5]    | `Integer.to_charlist/1`     | v1.3
+`Kernel.to_char_list/1`   | [v1.5]    | `Kernel.to_charlist/1`      | v1.3
+`String.to_char_list/1`   | [v1.5]    | `String.to_charlist/1`      | v1.3
+`t:char_list/0` typespec  | [v1.5]    | `t:charlist/0` typespec     | v1.3
+`Access.key/1`            | [v1.4]    | `Access.key/2`              | v1.3
+`Behaviour` module        | [v1.4]    | `@callback`                 | v1.0
+`Enum.uniq/2`             | [v1.4]    | `Enum.uniq_by/2`            | v1.2
+`Float.to_char_list/2`    | [v1.4]    | `:erlang.float_to_list/2`   | OTP v17.0
```

Comment:
We should refer to OTP as "OTP 17". WIth no `v` and no minor version because we don't want developers to depend on specific features based on minor OTP versions.